core/kotlinx-coroutines-core/test/TestBase.kt

```diff
@@ -105,13 +113,12 @@ public actual open class TestBase actual constructor() {
         finished.set(false)
     }
 
-    private lateinit var threadsBefore: Set<Thread>
-    private val SHUTDOWN_TIMEOUT = 10_000L // 10s at most to wait
-
     @Before
     fun before() {
         initPoolsBeforeTest()
         threadsBefore = currentThreads()
+        originalUncaughtExceptionHandler = Thread.getDefaultUncaughtExceptionHandler()
+        Thread.setDefaultUncaughtExceptionHandler({ t, e -> uncaughtExceptions.add(e) })
```

Comment:
It makes sense to print the stacktrace of this exception to the console (just like original handler does). This would help to trace a problem in case when the test hangs (also, the assertion at `after` does not print stacktraces either).